.github/workflows/test.yml

```
@@ -1,20 +1,42 @@
 name: test
 on:
   workflow_call:
+    secrets:
+      GH_PRIVATE_MODULES_TOKEN:
+        required: true
 
 jobs:
   test-short:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 #v5.0.0
 
+      - name: Configure Git for private modules
+        run: git config --global url."https://${{ secrets.GH_PRIVATE_MODULES_TOKEN }}@github.com/".insteadOf "https://github.com/"
+
       - uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00 #v6.0.0
+        id: go
         with:
           go-version-file: "go.mod"
+        env:
+          GOPRIVATE: github.com/celestiaorg/*
+
+      - name: Cache Go modules
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ steps.go.outputs.go-version }}-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-${{ steps.go.outputs.go-version }}-
+
+      - name: Download Go modules
+        run: go mod download
 
       - name: Run tests in short mode
         run: make test-short
-        timeout-minutes: 10
+        timeout-minutes: 25
```

seems like test run time for test-short should not have changed so proposal to revert or skip any tests that take longer than 10 minutes from test-short


```suggestion
        timeout-minutes: 10
```

---

yeah there's something weird with the private repo thing that this has to redownload all the go mods again and it can take a while

not sure why we're seeing this tbh